table.py

```diff
@@ -131,26 +145,37 @@ def delete_record(self, key):
     """
     def update_record(self, key, columns):
         # get the base record
-        record = self.read_record(key=key, query_columns=[0, 1, 2, 3])
+        record = self.read_record(key=key, query_columns=[x for x in range(0, len(columns) + 4)])
 
         # get the schema encoding
         schema_encoding = ''
         for i in columns:
-            # if value in column is 'None' add 0
+            # if value in column is not 'None' add 1qwe
             if columns[i]:
-                schema_encoding = schema_encoding + '0'
+                schema_encoding = schema_encoding + '1'
 
             # else add 1
             else:
-                schema_encoding = schema_encoding + '1'
+                schema_encoding = schema_encoding + '0'
 
         schema_encoding = int(schema_encoding, 2)
 
         # find value for indirection column
-        if record.columns[INDIRECTION_COLUMN]:
-            indirection_value = record.columns[INDIRECTION_COLUMN]
-        else:
-            indirection_value = None
+        # indirection_value = None #initialize
```

Comment:
excellent work here. fantastic catch. you can totally delete the commented out code.

The only issue is that you initialize "latest_rec_values" to None and then try to append to it. Can't append to None value as its not a list